src/Npgsql/TypeHandlers/CompositeHandlers/CompositeConstructorHandler.cs

```diff
@@ -41,11 +41,12 @@ public static CompositeConstructorHandler<TComposite> Create(PostgresType postgr
             if (parameterHandlers.Length > maxGenericParameters)
                 return new CompositeConstructorHandler<TComposite>(postgresType, constructorInfo, parameterHandlers);
 
-            var parameterTypes = new Type[maxGenericParameters + 1];
-            for (var parameterIndex = 0; parameterIndex < maxGenericParameters; ++parameterIndex)
-                parameterTypes[parameterIndex + 1] = parameterIndex < parameterHandlers.Length
-                    ? parameterHandlers[parameterIndex].ParameterInfo.ParameterType
-                    : typeof(Unused);
+            var parameterTypes = new Type[1 + maxGenericParameters];
```

Comment:
nit: make the element nullable (`Type?`) (this will probably require a bang below when passing it, but it's still more correct).

Reply:
`MakeGenericType` accepts an array of non-nullable elements, so it would be wrong.

Reply:
At the point where you're passing the array to MakeGenericType, you know it doesn't contains nulls anymore because you've filled them with `Unused` (that's what the second loop is about) - but the compiler can't know that. So the array is originally `Type?[]`, but later is known not to contain nulls, hence the bang.

Reply:
It gives no warning, so it's fine (:

Reply:
Well, the code isn't properly annotated for nullability, regardless of warnings. The IDE flags the `??= typeof(Unused)` as unneeded - because the array is annotated to never contain nulls. So someone may remove it by accident in the future.